### ISSUE TITLE
Using a setter on a property with list[TypeVar[Hashable]] reports error

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#4293]
**Describe the bug**
If you have an asymmetric property for a getter/setter, and the setter uses a `TypeVar` defined as bound to `Hashable`, the setter reports an error when used.

**To Reproduce**
```python
from typing import Hashable, TypeVar, Sequence

HashableT = TypeVar("HashableT", bound=Hashable)

class Foo:
    def __init__(self):
        self._something = []

    @property
    def something(self) -> Sequence[Hashable]:
        return self._something

    @something.setter
    def something(self, newthing: list[HashableT]):
        self._something = newthing

    def set_something(self, newthing: list[HashableT]):
        self._something = newthing


f = Foo()
f.something = ["a", "b", "c"]
f.set_something(["a", "b", "c"])
```

pyright reports on `f.something =  ["a", "b", "c"]` is invalid, while `f.set_something(["a", "b", "c"])` is valid:

Error message:
```text
prophash.py:23:3 - error: Cannot assign member "something" for type "Foo"
    Type "Literal['a']" cannot be assigned to type "HashableT@something"
    Type "Literal['b']" cannot be assigned to type "HashableT@something"
    Type "Literal['c']" cannot be assigned to type "HashableT@something" (reportGeneralTypeIssues)
```

**Expected behavior**
The setter should not report an error.

**VS Code extension or command-line**
Command line pyright 1.1.283, python 3.9

**Additional context**

An issue for `pandas-stubs` with using `DataFrame.columns` setter.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by ifuatgucluer** [comment]
I want to write cod like you but I startin stage at now. What can I do?

Eric Traut ***@***.***>, 21 Ara 2022 Çar, 10:27 tarihinde
şunu yazdı:

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.285, which I just published. It will also be included in a future release of pylance.
